test/jest/config.js

```diff
@@ -0,0 +1,26 @@
+const path = require('path');
+
+module.exports = {
+  collectCoverageFrom: [
+    'ui_framework/components/**/*.js',
+    // Seems to be a bug with jest or micromatch, in which the above glob
+    // doesn't match subsequent levels of directories, making this glob necessary.
+    'ui_framework/components/**/**/*.js',
+    '!ui_framework/components/index.js',
+    '!ui_framework/components/**/*/index.js',
+  ],
+  coverageDirectory: '<rootDir>/target/test_coverage',
```

Comment:
`target` seems like an odd place to put this -- maybe I'm not entirely clear on what this directory is meant for. We currently output other test-diagnostic information into `test`, e.g. `test/screenshots`, so how about we output the coverage report into `test/coverage`? We'd probably want to update the `.gitignore` in this case.